### ISSUE TITLE
Fix automated test failures 

### ISSUE THREAD

**Issue by geerlingguy** [geerlingguy/drupal-vm#1313]
### Issue Type

 - Bug Report

### Full console output

```
TASK [geerlingguy.drupal : Install Drupal with drush.] *************************
ERROR! Unexpected Exception: '0242ac11-0002-c14e-7964-000000002436'
to see the full traceback, use -vvv
```

### Summary

Tests are failing. That makes me sad.

Relevant commit: https://github.com/geerlingguy/drupal-vm/commit/e631891b922afdca4f1d1c3f916ae14ee391e787
Travis failure: https://travis-ci.org/geerlingguy/drupal-vm/builds/224132998

**Comment by geerlingguy** [comment]
The failures were introduce upon updating the Drupal role to `1.9.0`, so it's likely one of those changes (though that particular role is working okay in its own tests... so maybe it's an environment thing?).

**Comment by geerlingguy** [comment]
And of course it _yet again_ reminds me that I should always push a branch, even for what I think are inconsequential changes :P

**Comment by geerlingguy** [comment]
/me needs to give up on this for the day; worst case we can rollback the 1.9.0 drupal role update. But I'd like to find out what's up then push 1.9.1 soon. Note that it worked in my local testing against prod.drupalvm.com :/

**Comment by oxyc** [comment]
This is reproducible with a standard Vagrant setup.

**Comment by oxyc** [comment]
```
TASK [geerlingguy.drupal : Install Drupal with drush.] *************************
task path: /Users/cindy/Projects/Personal/drupal-vm/provisioning/roles/geerlingguy.drupal/tasks/install-site.yml:21
Using module file /usr/local/lib/python2.7/site-packages/ansible/modules/commands/command.py
<127.0.0.1> ESTABLISH SSH CONNECTION FOR USER: vagrant
<127.0.0.1> SSH: EXEC ssh -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ForwardAgent=yes -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=2200 -o 'IdentityFile="/Users/cindy/.vagrant.d/insecure_private_key"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=vagrant -o ConnectTimeout=30 -o ControlPath=/Users/cindy/.ansible/cp/17d6e21220 127.0.0.1 '/bin/sh -c '"'"'echo ~ && sleep 0'"'"''
<127.0.0.1> (0, '/home/vagrant\n', '')
<127.0.0.1> ESTABLISH SSH CONNECTION FOR USER: vagrant
<127.0.0.1> SSH: EXEC ssh -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ForwardAgent=yes -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=2200 -o 'IdentityFile="/Users/cindy/.vagrant.d/insecure_private_key"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=vagrant -o ConnectTimeout=30 -o ControlPath=/Users/cindy/.ansible/cp/17d6e21220 127.0.0.1 '/bin/sh -c '"'"'( umask 77 && mkdir -p "` echo /home/vagrant/.ansible/tmp/ansible-tmp-1492729470.64-124895571025560 `" && echo ansible-tmp-1492729470.64-124895571025560="` echo /home/vagrant/.ansible/tmp/ansible-tmp-1492729470.64-124895571025560 `" ) && sleep 0'"'"''
<127.0.0.1> (0, 'ansible-tmp-1492729470.64-124895571025560=/home/vagrant/.ansible/tmp/ansible-tmp-1492729470.64-124895571025560\n', '')
<127.0.0.1> PUT /var/folders/d8/8h53d56x645c5p32t6_44l_40000gn/T/tmpr6INPU TO /home/vagrant/.ansible/tmp/ansible-tmp-1492729470.64-124895571025560/command.py
<127.0.0.1> SSH: EXEC sftp -b - -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ForwardAgent=yes -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=2200 -o 'IdentityFile="/Users/cindy/.vagrant.d/insecure_private_key"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=vagrant -o ConnectTimeout=30 -o ControlPath=/Users/cindy/.ansible/cp/17d6e21220 '[127.0.0.1]'
<127.0.0.1> (0, 'sftp> put /var/folders/d8/8h53d56x645c5p32t6_44l_40000gn/T/tmpr6INPU /home/vagrant/.ansible/tmp/ansible-tmp-1492729470.64-124895571025560/command.py\n', '')
<127.0.0.1> ESTABLISH SSH CONNECTION FOR USER: vagrant
<127.0.0.1> SSH: EXEC ssh -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ForwardAgent=yes -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=2200 -o 'IdentityFile="/Users/cindy/.vagrant.d/insecure_private_key"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=vagrant -o ConnectTimeout=30 -o ControlPath=/Users/cindy/.ansible/cp/17d6e21220 127.0.0.1 '/bin/sh -c '"'"'chmod u+x /home/vagrant/.ansible/tmp/ansible-tmp-1492729470.64-124895571025560/ /home/vagrant/.ansible/tmp/ansible-tmp-1492729470.64-124895571025560/command.py && sleep 0'"'"''
<127.0.0.1> (0, '', '')
<127.0.0.1> ESTABLISH SSH CONNECTION FOR USER: vagrant
<127.0.0.1> SSH: EXEC ssh -o UserKnownHostsFile=/dev/null -o IdentitiesOnly=yes -o ForwardAgent=yes -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=2200 -o 'IdentityFile="/Users/cindy/.vagrant.d/insecure_private_key"' -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=vagrant -o ConnectTimeout=30 -o ControlPath=/Users/cindy/.ansible/cp/17d6e21220 -tt 127.0.0.1 '/bin/sh -c '"'"'/usr/bin/python /home/vagrant/.ansible/tmp/ansible-tmp-1492729470.64-124895571025560/command.py; rm -rf "/home/vagrant/.ansible/tmp/ansible-tmp-1492729470.64-124895571025560/" > /dev/null 2>&1 && sleep 0'"'"''
<127.0.0.1> (0, '\r\n{"changed": true, "end": "2017-04-20 23:05:00.498951", "stdout": "You are about to DROP all tables in your \'drupal\' database. Do you want to continue? (y/n): y", "cmd": ["/var/www/drupalvm/drupal/vendor/drush/drush/drush", "site-install", "standard", "-y", "--root=/var/www/drupalvm/drupal/web", "--site-name=Drupal", "--account-name=admin", "--account-pass=admin", "--db-url=mysql://drupal:drupal@127.0.0.1/drupal"], "rc": 0, "start": "2017-04-20 23:04:29.350447", "stderr": "Starting Drupal installation. This takes a while. Consider using the \\u001b[1;32;40m\\u001b[1m[ok]\\u001b[0m\\n--notify global option.\\nInstallation complete.                                               \\u001b[1;32;40m\\u001b[1m[ok]\\u001b[0m\\nCongratulations, you installed Drupal!                               \\u001b[1;32;40m\\u001b[1m[status]\\u001b[0m", "delta": "0:00:31.148504", "invocation": {"module_args": {"warn": true, "executable": null, "chdir": "/var/www/drupalvm/drupal/web", "_raw_params": "/var/www/drupalvm/drupal/vendor/drush/drush/drush site-install standard -y --root=/var/www/drupalvm/drupal/web --site-name=\\"Drupal\\" --account-name=admin --account-pass=admin --db-url=mysql://drupal:drupal@127.0.0.1/drupal ", "removes": null, "creates": null, "_uses_shell": false}}, "warnings": []}\r\n', 'Shared connection to 127.0.0.1 closed.\r\n')
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
skipping vars_file 'vars/{{ ansible_os_family }}.yml' due to an undefined variable
ERROR! Unexpected Exception: 'a8667f1f-4f6d-3768-9536-000000002438'
the full traceback was:

Traceback (most recent call last):
  File "/usr/local/bin/ansible-playbook", line 109, in <module>
    exit_code = cli.run()
  File "/usr/local/lib/python2.7/site-packages/ansible/cli/playbook.py", line 154, in run
    results = pbex.run()
  File "/usr/local/lib/python2.7/site-packages/ansible/executor/playbook_executor.py", line 153, in run
    result = self._tqm.run(play=play)
  File "/usr/local/lib/python2.7/site-packages/ansible/executor/task_queue_manager.py", line 284, in run
    play_return = strategy.run(iterator, play_context)
  File "/usr/local/lib/python2.7/site-packages/ansible/plugins/strategy/linear.py", line 278, in run
    results += self._wait_on_pending_results(iterator)
  File "/usr/local/lib/python2.7/site-packages/ansible/plugins/strategy/__init__.py", line 578, in _wait_on_pending_results
    results = self._process_pending_results(iterator)
  File "/usr/local/lib/python2.7/site-packages/ansible/plugins/strategy/__init__.py", line 444, in _process_pending_results
    if original_host not in self._notified_handlers[target_handler._uuid]:
KeyError: 'a8667f1f-4f6d-3768-9536-000000002438'
Ansible failed to complete successfully. Any error output should be
visible above. Please fix these errors and try again.
````

**Comment by oxyc** [comment]
One hacky workaround is

> If you replace your handlers with listen: instead of using name, error disappeared.
> https://github.com/ansible/ansible/issues/18411#issuecomment-280085162

**Comment by geerlingguy** [comment]
Fixed via merging #1316 — but will follow-up with a better fix in https://github.com/geerlingguy/drupal-vm/issues/1317
